snap/info.go

```diff
@@ -538,6 +559,8 @@ func ReadInfo(name string, si *SideInfo) (*Info, error) {
 		return nil, err
 	}
 
+	info.renameClashingCorePlugs()
```

Comment:
I think we need to revisit this in general as some "fixup" is done in `overlord/snapmgr` and some here but I didn't want to do any wider changes.